solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -221,6 +223,19 @@
   // only kept to identify if the cleaner has already been started.
   private Future<?> collectionPropsCacheCleaner;
 
+  private static BaseCloudSolrClient client;
```

Comment:
Important: this addition is definitely not cool.  The refactoring here shouldn't be adding state (fields on classes) to anything.